lib/httparty/request.rb

```diff
@@ -276,11 +276,10 @@ def get_charset
     end
 
     def encode_with_ruby_encoding(body, charset)
-      if !body.nil? && Encoding.name_list.include?(charset)
-        body.force_encoding(charset)
-      else
-        body
-      end
+      encoding = Encoding.find(charset)
+      body.force_encoding(charset)
+    rescue ArgumentError
```

Comment:
Ok, last thing I’ll post here. It looks like there were two different solutions to the `nil` body problem added within just a few days of each other, so I’d guess one was added without knowledge of the other:

1. May 12, the check inside `encode_with_ruby_encoding` and a test: 9bd77528fddf4f20d3f5ed749d340f535ee19765
2. May 16, just never call `encode_with_ruby_encoding` at all in this case: d721b76680c2b4936a8820d065da0039f851bbcb